src/req.cpp

```diff
@@ -95,8 +95,10 @@ int zmq::req_t::xsend (msg_t *msg_)
             id.set_flags (msg_t::more);
 
             rc = dealer_t::sendpipe (&id, &_reply_pipe);
-            if (rc != 0)
+            if (rc != 0) {
+                free (request_id_copy);
```

Comment:
thanks for the PR, but it's actually completely pointless to do a malloc manually here - it's just 4 bytes. Could you please instead change from init_data to init_size and then memcpy &_request_id into id.data() ? That way it will be only on the stack